pkg/asset/machines/master.go

```diff
@@ -236,3 +258,23 @@ func (m *Master) StructuredMachines() ([]machineapi.Machine, error) {
 
 	return machines, nil
 }
+
+// IsMasterManifest tests whether a file is a manifest that belongs to the
+// Master Machines asset.
+func IsMasterManifest(file *asset.File) bool {
+	if filepath.Dir(file.Filename) != directory {
+		return false
+	}
+	filename := filepath.Base(file.Filename)
+	if filename == masterUserDataFileName {
+		return true
+	}
+	if machineconfig.IsManifest(filename) {
+		return true
```

Comment:
Won't this slurp up worker MachineConfigs too?  I'd have expected to feed it a pattern or role string.

Reply:
fixed 8967ccd066030ebbb18fb3d19f7dd693df8abc46...c1ae01308bba2c83946b140c3fe5d740ad671a8f PTAL